### PR TITLE
Improve delete project endpoint logic and reduce UAC calls

### PR DESCRIPTION
**Pull request by anandjakhaniya**
Related Proto PR: https://github.com/VertaAI/modeldb/pull/490

### CODE REVIEW

**Comment by ravishetye** [comment]
@evgeniilezhnin : can you review please.

File: protos/protos/public/uac/RoleService.proto [hunk 1]
```diff
@@ -249,6 +249,14 @@ message DeleteRoleBinding {
     };
 };
 
+message DeleteRoleBindings {
```
Comment (ravishetye) [inline protos/protos/public/uac/RoleService.proto:252]: extract this in separate PR

File: backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java [hunk 2]
```diff
@@ -736,32 +735,33 @@ private void deleteRoleBindingsOfAccessibleProjects(List<ProjectEntity> allowedP
...
+          roleService.getResourceRoleBindings(
+              projectId, project.getOwner(), ModelDBServiceResourceTypes.PROJECT));
 
       // Delete workspace based roleBindings
-      deleteWorkspaceRoleBindings(
-          project.getWorkspace(),
-          WorkspaceType.forNumber(project.getWorkspace_type()),
-          project.getId(),
-          ProjectVisibility.forNumber(project.getProject_visibility()));
+      List<String> workspaceRoleBindingNames =
```
Comment (evgeniilezhnin) [inline backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java:750]: Can you also add this batch delete logic for datasets and repositories? (they have almost the same same code. For datasets it's a little different).

File: backend/src/test/java/ai/verta/modeldb/blobs/DiffAndMerge.java [hunk 3]
```diff
@@ -20,11 +20,11 @@ public void diffAndMerge(AutogenBlob a, AutogenBlob b) throws ModelDBException {
     AutogenBlobDiff d = DiffComputer.computeBlobDiff(newA, newB);
 
     // Applying the diff on top of the original A should get original B
-    AutogenBlob diffedB = DiffMerger.mergeBlob(newA, d);
+    AutogenBlob diffedB = DiffMerger.mergeBlob(newA, d, new HashSet<String>());
```
Comment (evgeniilezhnin) [inline backend/src/test/java/ai/verta/modeldb/blobs/DiffAndMerge.java]: I think this PR is already created for these tests fixes: https://github.com/VertaAI/modeldb/pull/484 @ravishetye

File: backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java (hunk 2 above)
Comment (evgeniilezhnin) [reply backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java:750]: and for create also

**Review by evgeniilezhnin — CHANGES_REQUESTED** [review]

File: backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java (hunk 2 above)
Comment (anandjakhaniya) [reply backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java:750]: Yes it is in my taks list and i will do it later in seperate PR

**Review by evgeniilezhnin — APPROVED** [review]

File: protos/protos/public/uac/RoleService.proto (hunk 1 above)
Comment (anandjakhaniya) [reply protos/protos/public/uac/RoleService.proto:252]: extracted PR: https://github.com/VertaAI/modeldb/pull/490

File: backend/src/test/java/ai/verta/modeldb/blobs/DiffAndMerge.java (hunk 3 above)
Comment (anandjakhaniya) [reply backend/src/test/java/ai/verta/modeldb/blobs/DiffAndMerge.java]: reverted

File: backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java (hunk 2 above)
Comment (ravishetye) [reply backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java:750]: track it in jira please
